pkg/ottl/ottlfuncs/func_replace_all_matches.go

```diff
@@ -75,11 +76,14 @@ func replaceAllMatches[K any](target ottl.PMapGetter[K], pattern string, replace
 				return nil, err
 			}
 		}
-		for _, value := range val.All() {
+
+		updated := pcommon.NewMap()
```

Comment:
The temporary map `updated` has been introduced here to avoid changing the input map `val`.
I have no `urban knowledge` here, so please advise whether
a) it's ok to change `val` here and `updated` should be removed
or
b) whether other functions should also not change `val`

Reply:
Then we should at least take care that the OTTL functions doesn't change the values to avoid unexpected side-effects. We should create an own issue for this for further discussion.

Reply:
I removed this piece of code again, we can discuss this separately when it pops up in reality.